.github/scripts/rollout_healthcheck.sh

```diff
@@ -0,0 +1,375 @@
+    oc login --token=$OC_TEMP_TOKEN --server=$OC_SERVER
+    oc project $OC_NAMESPACE # Safeguard!
+fi
+
+get_workload_list() {
+    local label_selector=$1
+    local workload_list
+    workload_list=$(oc get all -n $OC_NAMESPACE -l $label_selector -oname)
+    echo "$workload_list"
+}
+
+echo_red() {
+    echo -e "\033[0;31m$1\033[0m"
+}
+
+echo_green() {
+    echo -e "\033[0;32m$1\033[0m"
+}
+
+echo_yellow() {
+    echo -e "\033[0;33m$1\033[0m"
+}
+
+echo_checkmark() {
+    echo  "✅"
+}
+
+echo_cross() {
+    echo "❌"
+}
+
+# checks if the deployment has replicas in the correct states
+_health_check_deployment() {
+    local healthy="false"
+    local no_unready_replicas="false"
+    local equal_ready_replicas="false"
+    local one_available_replica="false"
+    local workload_name=$1
+    local replicas
+    local unavailable_replicas
+    local ready_replicas
+    replicas=$(oc get -n $OC_NAMESPACE $workload_name -o jsonpath='{.status.replicas}')
+    # 2 following keys can not exist in the manifest if they're 0
+    unavailable_replicas=$(oc get -n $OC_NAMESPACE $workload_name -o jsonpath='{.status.unavailableReplicas}')
+    one_available_replica=$(oc get -n $OC_NAMESPACE $workload_name -o jsonpath='{.status.availableReplicas}')
+    if [ -z "$unavailable_replicas" ]; then
+        unavailable_replicas=0
+    fi
+    ready_replicas=$(oc get -n $OC_NAMESPACE $workload_name -o jsonpath='{.status.readyReplicas}')
+    if [ -z "$ready_replicas" ]; then
+        ready_replicas=0
+    fi
+    if [ -z "$one_available_replica" ]; then
+        one_available_replica=0
+    fi
+    # begin replica flag setting
+    if [ "$replicas" -eq "$ready_replicas" ]; then
+        equal_ready_replicas="true"
+    fi
+    if [ "$unavailable_replicas" -eq 0 ]; then
+        no_unready_replicas="true"
+    fi
+    if [ "$one_available_replica" -eq 1 ]; then
+        one_available_replica="true"
+    fi
+    if [ "$equal_ready_replicas" == "true" ] && [ "$no_unready_replicas" == "true" ] && [ "$one_available_replica" == "true" ]; then
+        healthy="true"
+    fi
+    echo "$healthy"
+}
+
+# checks if a pod is running
+_pod_running() {
+    local pod_name=$1
+    local ready="false"
+    local pod_status
+    pod_status=$(oc get -n $OC_NAMESPACE $pod_name -o jsonpath='{.status.phase}')
+    if [ "$pod_status" == "Running" ]; then
+        ready="true"
+    fi
+    echo "$ready"
+}
+
+# polls deployments and pods for readiness before kicking off triage info collection
+poll_deployments() {
+    local succeeded="true"
+    local deployment_list=$1
+    local pod_list=$2
+    local deployment_status
+    for deployment in $deployment_list; do
+        deployment_status=$(_health_check_deployment $deployment)
+        if [ "$deployment_status" == "false" ]; then
+            succeeded="false"
+            break
+        fi
+    done
+    for pod in $pod_list; do
+        _pod_running=$(_pod_running $pod)
+        if [ "$pod_status" == "false" ]; then
```

Comment:
Is this correct?  Is pod_status available in this context when it's deployed as a local variable in the _pod_running method?

Why not just if [ " (_pod_running $pod)" == "false ]